### ISSUE TITLE
python novas support for newer ephemerides

### ISSUE THREAD

**Issue by randyg3000** [brandon-rhodes/python-novas#16]
Hello!

There does not seem to be support for newer than de405.bsp. Having previously done

>pip install novas_de405

(works with ephem_opn())

I tried forcing the issue by downloading de440.bsp and then

>jd_begin, jd_end, de_number = ephem_open('de440.bsp')

which gives

File "/usr/local/lib/python3.6/site-packages/novas/compat/eph_manager.py", line 90, in ephem_open
    ctypes.byref(de_number))
  File "/usr/local/lib/python3.6/site-packages/novas/__init__.py", line 53, in _check_c_errors
    raise error[0](error[1])
OSError: from C function 'ephem_open': unable to set record length; ephemeris (DE number) not in look-up table.

Side note: "skyfield" is able to open the de440.bsp (eph = api.load("de440.bsp"))

**Comment by brandon-rhodes** [comment]
That's correct — the NOVAS library has no support for ephemerides beyond those that are hard-coded in its C code:

https://github.com/brandon-rhodes/python-novas/blob/2e690e925e53ed585f57b3105cf92ea70954b60e/Cdist/eph_manager.c#L230

You will, alas, have to contact the USNO to see if they ever plan on updating NOVAS to support newer ephemerides; this GitHub repository is merely a copy of the code from their web site (which I think has now disappeared?) with one or two fixes to at least let it compile under modern Python. It's possible that the library is now more or less abandoned. I copied it here to GitHub primarily so I could run its routines as a double-check against most modern Python astronomy libraries; I don't anticipate adding new features independent of the work of the USNO itself, lest this go out of sync with any development they might do on the library in the future.

(Feel free to comment further here about your project and its needs, even though I'm going to close the issue since this repository is merely a copy of the project and not the real project itself. Thanks!)

**Comment by randyg3000** [comment]
Hi Brandon,

Thank you, that is very helpful. I've thought of NOVAS as being definitive, but I agree it has received no attention now for quite a while. I've been able to match NOVAS to skyfield to sub-milli-arcseconds (for stars) except for high PPM stars e.g. Barnard where they match to ~10-15 mas. Could you comment on what you feel is the definitive "modern Python astronomy" library? I'm not sure if skyfield is that very thing, or whether you depend on other libraries? Our project requires calculating star position az/el (without refraction) to the utmost accuracy. That extent I'm downloading finals2000A and installing the polar motion table to set up for:

skyfield_azel = also_here.at(t2).observe(star_for_skyfield).apparent()
salt, saz, distance = skyfield_azel.altaz()  # (no refraction)

and this matches NOVAS as I've described, above.

**Comment by brandon-rhodes** [comment]
Skyfield has traditionally been the higher-precision alternative, as it was designed to perform the same floating point operations as NOVAS; but it does have fewer features than AstroPy. Many of AstroPy’s problems with precision have been fixed, so you might want to try them both out on a small sample problem of a few lines long and see which agrees better with a known authority, if you have one for your project (NOVAS? HORIZON?).

My favorite is Skyfield, of course, but I'm biased. :)
